scripts/metadata_comparison/extractor.py

```diff
@@ -0,0 +1,193 @@
+                api = 'lifesciences' if 'beta' in attempt.get('backend', '').lower() else 'genomics'
+                papi_operation_to_api_mapping[jobId] = api
+    return papi_operation_to_api_mapping
+
+
+def read_genomics_papiv1_operations_metadata(jobId, api, genomics_client):
+    """Reads the operations metadata for a pipelines API v1 job."""
+    return genomics_client.operations().get(name=jobId).execute()
+
+
+def pretty_print_json(raw_json):
+    return json.dumps(json.loads(raw_json), indent=4, separators=(',', ': '))
+
+
+def read_operations_metadata(jobId, api, genomics_client):
+    """Reads the operations metadata for a pipelines API v2alpha1 job ID. Returns a python dict"""
+    # TODO: Reroute appropriately for different API versions
+    print('Reading operation metadata for {jobId}...'.format(jobId=jobId), end='')
+    result = genomics_client.projects().operations().get(name=jobId).execute()
+    print(' done!')
+    return result
+
+
+def upload_blob(bucket_name, source_file_contents, destination_blob_name, gcs_storage_client):
+    """Uploads a file to the cloud"""
+    # bucket_name = "your-bucket-name"
+    # source_file_contents = "... some file contents..."
+    # destination_blob_name = "storage/object/name"
+
+    bucket = gcs_storage_client.bucket(bucket_name)
+
+    print('Uploading file content to gs://{bucket}/{dest}...'.format(bucket=bucket_name, dest=destination_blob_name), end='')
+    blob = bucket.blob(destination_blob_name)
+    blob.upload_from_string(source_file_contents)
+    print(" done!")
+
+
+def upload_workflow_metadata_json(bucket_name, raw_workflow_metadata, workflow_gcs_base_path, gcs_storage_client):
+    workflow_gcs_metadata_upload_path = '{workflow_gcs_base_path}/metadata.json'.format(workflow_gcs_base_path=workflow_gcs_base_path)
+    upload_blob(bucket_name, raw_workflow_metadata, workflow_gcs_metadata_upload_path, gcs_storage_client)
+
+
+def upload_operations_metadata_json(bucket_name, operation_id, operations_metadata, workflow_gcs_base_path, gcs_storage_client):
+    """Uploads metadata to cloud storage, as json"""
+    operation_upload_path = '{workflow_gcs_base_path}/operations/{miniid}.json'.format(workflow_gcs_base_path=workflow_gcs_base_path, miniid=get_operation_id_number(operation_id))
+    formatted_metadata = json.dumps(operations_metadata, indent=2)
+    upload_blob(bucket_name, formatted_metadata, operation_upload_path, gcs_storage_client)
+
+
+def process_workflow(cromwell_url, gcs_bucket, gcs_path, gcs_storage_client, genomics_client, workflow):
+    raw_metadata, json_metadata = fetch_raw_workflow_metadata(cromwell_url, workflow)
+    workflow_gcs_base_path = '{root_gcs_path}/{wfid}/extractor'.format(root_gcs_path=gcs_path, wfid=workflow)
+
+    operation_ids = find_operation_ids_in_metadata(json_metadata)
+    operation_metadata = { id: read_operations_metadata(id, api, genomics_client) for (id, api) in operation_ids.items() }
+    for id in operation_metadata:
```

Comment:
does this need to be looping twice and making that dictionary with an entry for every operation?